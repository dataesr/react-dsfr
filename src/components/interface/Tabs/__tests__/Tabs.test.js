import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import { Tabs, Tab } from '../index';

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

Enzyme.configure({ adapter: new Adapter() });

describe('<Tabs />', () => {
  const initProps = {
    className: 'my-class',
    scheme: 'soft-yellow-medium',
  };
  let wrapper;

  beforeEach(() => {
    uuidv4.mockImplementation(() => 'xxxxxxx');
    wrapper = shallow(
      <Tabs
        className={initProps.className}
        scheme={initProps.scheme}
        defaultActiveTab={1}
        data-testid="tabs"
      >
        <Tab label="Label" data-test-id="tab">Tab</Tab>
      </Tabs>,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Tabs properly', () => {
    const component = renderer
      .create(
        <Tabs scheme="soft-blue-soft">
          <Tab label="Label1">Tab #1</Tab>
          <Tab label="Label2">Tab #2</Tab>
        </Tabs>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should have custom class', () => {
    expect(wrapper.find('.fr-tabs').hasClass('my-class')).toBeTruthy();
  });

  it('should call setHeight', () => {
    const setHeight = jest.fn();
    const useState = jest.spyOn(React, 'useState');
    useState.mockImplementation((contentTabHeight) => [contentTabHeight, setHeight]);
    expect(setHeight).toBeTruthy();
  });
});
