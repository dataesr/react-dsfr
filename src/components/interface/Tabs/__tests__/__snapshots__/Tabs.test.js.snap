// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render Tabs properly 1`] = `
<div
  className="fr-tabs fr-scheme-soft-blue-soft"
  style={
    Object {
      "height": 500,
    }
  }
>
  <ul
    className="fr-tabs__list"
    role="tablist"
  >
    <li
      role="presentation"
    >
      <button
        aria-controls="fr-tabpanel-0"
        aria-selected="true"
        className="fr-tabs__tab"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex="0"
        type="button"
      >
        
        Label1
      </button>
    </li>
    <li
      role="presentation"
    >
      <button
        aria-controls="fr-tabpanel-1"
        aria-selected="false"
        className="fr-tabs__tab"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex="0"
        type="button"
      >
        
        Label2
      </button>
    </li>
  </ul>
  <div
    className="fr-tabs__panel fr-tabs__panel--selected"
    id="fr-tabpanel-xxxxxxx"
    role="tabpanel"
    tabIndex="0"
  >
    Tab #1
  </div>
  <div
    className="fr-tabs__panel "
    id="fr-tabpanel-xxxxxxx"
    role="tabpanel"
    tabIndex="0"
  >
    Tab #2
  </div>
</div>
`;
